### DISCUSSION TITLE
Discretisation of Darcy Equation with variable coefficients

### DISCUSSION THREAD

**Discussion by annereinarz** [wenceorg/comp4187#91] [General]
For anyone with issues getting the 7 point stencil for the Darcy equation:

- There are several valid approaches for doing this, the first (but possibly not best) approach is to assume that the coefficient k is locally constant and simply moving it out of the derivative. Then at each point you apply the 7 point stencil for the laplace operator and multiple with the value of k at that point.
- The second (better) approach is to compute the derivative of the coefficient k analytically. You start by applying the product rule
$$\nabla \cdot (k(y) \nabla) = (\nabla k(y)) \cdot \nabla + k(y) \Delta$$ 
Now you can compute $\nabla k(y)$, since it depends only on y the first and last component are zero, in total we get:
$$ k'(y) d_y + k(y) (d_x^2+d_y^2+d_z^2)$$
where $k'$ is the analytically computed derivative of k.

You can then add the 7 point stencil for the laplacian (multiplied with k) to a centered difference stencil for $d_y$

**Comment by emboyer** [comment]
Presumably the one sided difference doesn't also need to be 7 points, and why does that not reduce the overall convergence rate?

**Reply by annereinarz** [reply]
The stencil for the $d_y$ component doesn't need any point in x or z direction. But indeed you are right, the one-sided difference would give you a first order accurate scheme. If you want to retain the second order accuracy you would need a centered stencil there. (Note: I've updated this in the snippet above)
